### PR TITLE
crystal: 0.31.1

### PR DESCRIPTION
**Pull request by bcardiff**
There is a new patch release of Crystal.
This release allows moving to the upstream patch of an upcoming feature of bdw-gc.

So in order to remove the custom build of gc for Crystal instroduced in #39026, an update to the bdw-gc formula is needed.

The bdw-gc patch was merged in https://github.com/ivmai/bdwgc/pull/277 and [it should be released in 8.2.x](https://github.com/ivmai/bdwgc/issues/274#issuecomment-489634982)

---

`brew audit --strict crystal` is complaining locally about llvm@8 been an alias, but it's no longer an alias.

### CODE REVIEW

File: Formula/bdw-gc.rb [hunk 1]
```diff
@@ -21,6 +22,12 @@ class BdwGc < Formula
   depends_on "libatomic_ops" => :build
   depends_on "pkg-config" => :build
 
+  patch do
+    # Required for v8.0.4. Support for green threads.
+    url "https://github.com/ivmai/bdwgc/commit/5668de71107022a316ee967162bc16c10754b9ce.patch"
```
Comment (chenrui333) [inline Formula/bdw-gc.rb]:
```suggestion
    url "https://github.com/ivmai/bdwgc/commit/5668de71107022a316ee967162bc16c10754b9ce.patch?full_index=1"
```

**Comment by chenrui333** [comment]
Looks like the build failed due to the `full_index` thing.

**Comment by bcardiff** [comment]
Change with updated sha256 pushed @chenrui333

**Comment by fxcoudert** [comment]
> upstream patch of an upcoming feature of bdw-gc

It still seems like a large patch to backport, and could possibly introduce other issues…

**Comment by bcardiff** [comment]
@fxcoudert If you think that the [gc patch](https://raw.githubusercontent.com/crystal-lang/distribution-scripts/ab683792f34c60159f0e697adf792ff5b0fcbf91/linux/files/feature-thread-stackbottom.patch) is large the alternative would be to keep the gc vendored in crystal as before with the updated patch.

I happy to change to whatever you think is best for homebrew ecosystem. 

Note that the gc patch is adding a feature without affecting existing ones.

**Comment by bcardiff** [comment]
I pushed the change that keeps gc vendored. Let me know if there is something different that it would be preferred.

**Review by chenrui333 — APPROVED** [review]
lgtm

**Comment by chenrui333** [comment]
cc @fxcoudert @SMillerDev

**Review by fxcoudert — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @bcardiff for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**